rcl/src/rcl/wait.c

```diff
@@ -614,6 +614,154 @@ rcl_wait(rcl_wait_set_t * wait_set, int64_t timeout)
   return RCL_RET_OK;
 }
 
+rcl_ret_t
+rcl_wait_multiple(rcl_wait_set_t wait_sets[], const size_t num_wait_sets, int64_t timeout)
+{
+  RCL_CHECK_ARGUMENT_FOR_NULL(wait_sets, RCL_RET_INVALID_ARGUMENT);
+  size_t i;
+  size_t j;
+  rcl_wait_set_t * wait_set;
+  // Count number of entities in wait sets while checking for invalid wait sets
+  size_t number_of_subscriptions = 0u;
+  size_t number_of_guard_conditions = 0u;
+  size_t number_of_timers = 0u;
+  size_t number_of_clients = 0u;
+  size_t number_of_services = 0u;
+  for (i = 0u; i < num_wait_sets; ++i) {
```

Comment:
@jacobperron it could be that no wait set was provided, should we have a different rcl error message for that case (right now it'll say that wait sets are empty)?

Reply:
My intention was to return `RCL_RET_WAIT_SET_EMPTY` thinking it is intuitive, but I can add a new return code e.g. `RCL_RET_WAIT_SET_NONE`.

Reply:
Either that or `RCL_RET_INVALID_ARGUMENT` plus `RCL_SET_ERROR_MSG()` with a nice message.